src/util/routerinfo.cc

```diff
@@ -77,157 +77,139 @@ bool RouterInfoCommand::Impl(
+                routerInfo.GetCaps() | core::RouterInfo::Cap::HighBandwidth);
+          // Set options
+          routerInfo.SetOption("netId", std::to_string(vm["netid"].as<int>()));
+          routerInfo.SetOption("router.version", I2P_VERSION);
+          routerInfo.CreateBuffer(keys);
+
+          // Set filename if none provided
+          if (filename.empty())
+            filename = std::string("routerInfo-")
+                       + routerInfo.GetIdentHash().ToBase64()
+                       + std::string(".dat");
+          // Write key to file
+          core::OutputFileStream output_key(
+              filename + ".key", std::ofstream::binary);
+          if (output_key.Fail())
+            {
+              LOG(error) << "routerinfo: Failed to open file " << filename
+                         << ".key";
+              return false;
+            }
+          const std::size_t len = keys.GetFullLen();
+          std::unique_ptr<std::uint8_t[]> buf(
+              std::make_unique<std::uint8_t[]>(len));
+          keys.ToBuffer(buf.get(), len);
+          output_key.Write(buf.get(), len);
+          if (output_key.Fail())
+            {
+              LOG(error) << "routerinfo: Failed to write to file " << filename
+                         << ".key";
+              return false;
+            }
+          // Write RI to file
           routerInfo.SaveToFile(filename);
+          return true;
         }
-      catch (...)
-        {
-          core::Exception ex(GetName().c_str());
-          ex.Dispatch(__func__);
-          return false;
-        }
-
-      return true;
-    }
-
-  if (inputs.size() == 0)
-    {
-      LOG(error) << "routerinfo: Not enough arguments";
-      PrintUsage(cmd_name);
-      return false;
-    }
 
-  // for each file : read and print description
-  for (const auto& arg : inputs)
-    {
-      core::InputFileStream input(arg, std::ios::in | std::ios::binary);
-      if (input.Fail())
-        {
-          LOG(error) << "routerinfo: Failed to open input " << arg;
-          return false;
-        }
-      // Read all input
-      std::size_t length(0);
-      std::unique_ptr<std::uint8_t[]> buffer =
-          input.ReadAll<std::uint8_t, std::size_t>(&length);
-      if (!buffer)
+      if (inputs.size() == 0)
         {
-          LOG(error) << "routerinfo: Failed to read input " << arg;
+          LOG(error) << "routerinfo: Not enough arguments";
+          PrintUsage(cmd_name);
           return false;
         }
-      LOG(trace) << "routerinfo: read OK length " << length;
-      try
+
+      // for each file : read and print description
+      for (const auto& arg : inputs)
         {
+          core::InputFileStream input(arg, std::ios::in | std::ios::binary);
+          if (input.Fail())
+            {
+              LOG(error) << "routerinfo: Failed to open input " << arg;
+              return false;
+            }
+          // Read all input
+          std::size_t length(0);
```

Comment:
JFTR, since c++11:

>If T is a scalar type, the object's initial value is the integral constant zero explicitly converted to T.

So, `length()` or `length{}` should also work.